src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceGenerator.cs

```diff
@@ -48,72 +47,71 @@ public sealed class SourceGenerator : CachingSourceGenerator
             defaultSeverity: DiagnosticSeverity.Error,
             isEnabledByDefault: true);
 
-        protected override bool TryGetRelevantInput(in GeneratorExecutionContext context, out AdditionalText? input, out SourceText? inputText)
+        public void Initialize(IncrementalGeneratorInitializationContext context)
         {
-            input = context.AdditionalFiles.SingleOrDefault(a => Path.GetFileName(a.Path) == "Syntax.xml");
-            if (input == null)
-            {
-                context.ReportDiagnostic(Diagnostic.Create(s_MissingSyntaxXml, location: null));
-                inputText = null;
-                return false;
-            }
+            var syntaxXmlFiles = context.AdditionalTextsProvider.Where(at => Path.GetFileName(at.Path) == "Syntax.xml").Collect();
```

Comment:
You don't need to bother with the `.Collect()` you can just keep it as a ValuesProvider and pass that to the `RegisterSourceOutput`. That will get called for reach additional file in the collection, but we know there is only one, so it's fine.